src/utils/power_of_two.hh

```diff
@@ -0,0 +1,28 @@
+#ifndef POWER_OF_TWO_HH
+#define POWER_OF_TWO_HH
+
+#include "narrow.hh"
+#include <bit>
+#include <cassert>
+
+namespace openmsx {
+
+template<typename T>
+struct power_of_two {
+	constexpr power_of_two(T size)
+		: exponent(narrow_cast<uint8_t>(std::bit_width(size - 1)))
+	{
+		assert(std::has_single_bit(size));
+	};
+
+	[[nodiscard]] constexpr T operator()() const
```

Comment:
I don't know if this is a good idea or not, but have you considered instead of using `operator()`, to use an (implicit) conversion operator?
`[[nodiscard]] constexpr operator T() const { ...`

Reply:
Oh, I didn’t know that was possible. I’ll change that for sure.